libraries/chain/controller.cpp

```diff
@@ -4802,6 +4808,13 @@ struct controller_impl {
       return pending->active_producers();
    }
 
+   const producer_authority_schedule& scheduled_active_producers_at(block_timestamp_type t)const {
+      if( !(pending) )
+         return head_compute_active_schedule_auth(t);
+
+      return pending->active_producers();
```

Comment:
This doesn't seem right, why is `get_scheduled_active_proposer_policy_at` not used?

Reply:
If `pending` does not exist, we call `head_compute_active_schedule_auth` which in turns calls `get_scheduled_active_proposer_policy_at`. (I should have changed `head_compute_active_schedule_auth` to `head_scheduled_active_schedule_auth_at`).

If `pending` exists, we call `pending`'s `active_producers()`, which uses actual computed active policy.